### ISSUE TITLE
[Bug]: CPU backend makes mask conversion from ttir to ttsharedir wrong

### ISSUE THREAD

**Issue by ikushare** [microsoft/triton-shared#309]
### Triton python code

```mlir
import torch
import triton
import triton.language as tl
import benchmark

@triton.jit
def demo_kernel(x_ptr,y_ptr):
    rows = tl.program_id(0) +  tl.arange(0, 1)[:,None]
    cols = tl.arange(0, 8)[None, :]
    mask = (rows<2) & (cols<8)
    x = tl.load(x_ptr + rows*8+ cols, mask=mask)
    tl.store(y_ptr + rows*8+ cols, x, mask=mask)

#@benchmark.measure()
def bench_demo(size, provider):
    torch.manual_seed(0)
    a = torch.rand(size, device='cpu', dtype=torch.float32)
    b = torch.rand(size, device='cpu', dtype=torch.float32)
    if provider == 'torch':
        b = a
    if provider == 'triton':
        demo_kernel[(1,)](a, b,)
    return b

if __name__ == "__main__":
    benchmark.select_cpu_backend()
    size=(2,8)
    output_torch = bench_demo(size, 'torch')
    output_triton = bench_demo(size, 'triton')
    print(f"output_torch:{output_torch}")
    print(f"output_triton:{output_triton}")
    print(f'The maximum difference between torch and triton is '
          f'{torch.max(torch.abs(output_torch - output_triton))}')
```

### Triton IR

```mlir
#loc = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0)
module {
  tt.func public @demo_kernel(%arg0: !tt.ptr<f32> {tt.divisibility = 16 : i32} loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0), %arg1: !tt.ptr<f32> {tt.divisibility = 16 : i32} loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0)) attributes {noinline = false} {
    %c8_i32 = arith.constant 8 : i32 loc(#loc1)
    %c2_i32 = arith.constant 2 : i32 loc(#loc1)
    %cst = arith.constant dense<8> : tensor<1x8xi32> loc(#loc1)
    %0 = tt.get_program_id x : i32 loc(#loc2)
    %1 = tt.make_range {end = 8 : i32, start = 0 : i32} : tensor<8xi32> loc(#loc3)
    %2 = tt.expand_dims %1 {axis = 0 : i32} : tensor<8xi32> -> tensor<1x8xi32> loc(#loc4)
    %3 = arith.cmpi slt, %0, %c2_i32 : i32 loc(#loc5)
    %4 = arith.cmpi slt, %2, %cst : tensor<1x8xi32> loc(#loc6)
    %5 = tt.splat %3 : i1 -> tensor<1x8xi1> loc(#loc16)
    %6 = arith.andi %5, %4 : tensor<1x8xi1> loc(#loc7)
    %7 = arith.muli %0, %c8_i32 : i32 loc(#loc8)
    %8 = tt.addptr %arg0, %7 : !tt.ptr<f32>, i32 loc(#loc9)
    %9 = tt.splat %8 : !tt.ptr<f32> -> tensor<1x8x!tt.ptr<f32>> loc(#loc17)
    %10 = tt.addptr %9, %2 : tensor<1x8x!tt.ptr<f32>>, tensor<1x8xi32> loc(#loc10)
    %11 = tt.load %10, %6 : tensor<1x8x!tt.ptr<f32>> loc(#loc11)
    %12 = tt.addptr %arg1, %7 : !tt.ptr<f32>, i32 loc(#loc12)
    %13 = tt.splat %12 : !tt.ptr<f32> -> tensor<1x8x!tt.ptr<f32>> loc(#loc18)
    %14 = tt.addptr %13, %2 : tensor<1x8x!tt.ptr<f32>>, tensor<1x8xi32> loc(#loc13)
    tt.store %14, %11, %6 : tensor<1x8x!tt.ptr<f32>> loc(#loc14)
    tt.return loc(#loc15)
  } loc(#loc)
} loc(#loc)
#loc1 = loc(unknown)
#loc2 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":8:25)
#loc3 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":9:24)
#loc4 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":9:27)
#loc5 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":10:17)
#loc6 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":10:28)
#loc7 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":10:23)
#loc8 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":11:29)
#loc9 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":11:24)
#loc10 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":11:32)
#loc11 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":11:16)
#loc12 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":12:21)
#loc13 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":12:29)
#loc14 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":12:35)
#loc15 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":12:4)
#loc16 = loc(fused[#loc7, #loc5])
#loc17 = loc(fused[#loc10, #loc9])
#loc18 = loc(fused[#loc13, #loc12])
```

### Crash log

```shell
output_torch:tensor([[0.4963, 0.7682, 0.0885, 0.1320, 0.3074, 0.6341, 0.4901, 0.8964],
        [0.4556, 0.6323, 0.3489, 0.4017, 0.0223, 0.1689, 0.2939, 0.5185]])
output_triton:tensor([[0.4963, 0.8000, 0.1610, 0.2823, 0.6816, 0.9152, 0.3971, 0.8742],
        [0.4194, 0.5529, 0.9527, 0.0362, 0.1852, 0.3734, 0.3051, 0.9320]])
The maximum difference between torch and triton is 0.6038446426391602
```

### Additional information

ttsharedir：
```
#loc = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0)
module {
  func.func @demo_kernel(%arg0: memref<*xf32> {tt.divisibility = 16 : i32} loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0), %arg1: memref<*xf32> {tt.divisibility = 16 : i32} loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0), %arg2: i32 loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0), %arg3: i32 loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0), %arg4: i32 loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0), %arg5: i32 loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0), %arg6: i32 loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0), %arg7: i32 loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":7:0)) {
    %c8_i32 = arith.constant 8 : i32 loc(#loc1)
    %0 = arith.muli %arg5, %c8_i32 : i32 loc(#loc2)
    %1 = arith.index_cast %0 : i32 to index loc(#loc3)
    %reinterpret_cast = memref.reinterpret_cast %arg0 to offset: [%1], sizes: [1, 8], strides: [8, 1] : memref<*xf32> to memref<1x8xf32, strided<[8, 1], offset: ?>> loc(#loc4)
    %alloc = memref.alloc() : memref<1x8xf32> loc(#loc5)
    %subview = memref.subview %reinterpret_cast[0, 0] [1, 1] [1, 1] : memref<1x8xf32, strided<[8, 1], offset: ?>> to memref<1x1xf32, strided<[8, 1], offset: ?>> loc(#loc5)
    %subview_0 = memref.subview %alloc[0, 0] [1, 1] [1, 1] : memref<1x8xf32> to memref<1x1xf32, strided<[8, 1]>> loc(#loc5)
    memref.copy %subview, %subview_0 : memref<1x1xf32, strided<[8, 1], offset: ?>> to memref<1x1xf32, strided<[8, 1]>> loc(#loc5)
    %2 = bufferization.to_tensor %alloc restrict writable : memref<1x8xf32> to tensor<1x8xf32> loc(#loc5)
    %reinterpret_cast_1 = memref.reinterpret_cast %arg1 to offset: [%1], sizes: [1, 8], strides: [8, 1] : memref<*xf32> to memref<1x8xf32, strided<[8, 1], offset: ?>> loc(#loc6)
    %extracted_slice = tensor.extract_slice %2[0, 0] [1, 1] [1, 1] : tensor<1x8xf32> to tensor<1x1xf32> loc(#loc7)
    %subview_2 = memref.subview %reinterpret_cast_1[0, 0] [1, 1] [1, 1] : memref<1x8xf32, strided<[8, 1], offset: ?>> to memref<1x1xf32, strided<[8, 1], offset: ?>> loc(#loc7)
    bufferization.materialize_in_destination %extracted_slice in writable %subview_2 : (tensor<1x1xf32>, memref<1x1xf32, strided<[8, 1], offset: ?>>) -> () loc(#loc7)
    return loc(#loc)
  } loc(#loc)
} loc(#loc)
#loc1 = loc(unknown)
#loc2 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":11:29)
#loc3 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":12:21)
#loc4 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":11:32)
#loc5 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":11:16)
#loc6 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":12:29)
#loc7 = loc("/data/home24/duyb/triton_shared/python/examples/test_triu_cpu.py":12:35)
```

Error in “%subview = memref.subview %reinterpret_cast[0, 0] [1, 1] [1, 1] : memref<1x8xf32, strided<[8, 1], offset: ?>> to memref<1x1xf32, strided<[8, 1], offset: ?>> loc(#loc5)”,The size variable in %reinterpret_cast should not be [1,1]; it is expected to be [1,8]
Analysis found that when processing the mask of the load instruction in ttir, the parsing process of “%5 = tt.splat %3 : i1 -> tensor<1x8xi1>” records %3 as scalar, causing an error. The code location is: lib/Analysis/MaskAnalysis.cpp. The code is expected to execute minStates, but it actually executes minStateScalar.

```
LogicalResult MaskState::parseAnd(arith::AndIOp andOp, const Location loc,
                                  OpBuilder &builder) {
  assert(this->isEmpty());

  MaskState lhsState;
  if (failed(lhsState.parse(andOp.getLhs(), loc, builder)))
    return failure();

  MaskState rhsState;
  if (failed(rhsState.parse(andOp.getRhs(), loc, builder)))
    return failure();

  if(!lhsState.isMask() || !rhsState.isMask()) {
    return this->minStateScalar(lhsState, rhsState, loc, builder);
  }
  return this->minStates(lhsState, rhsState, loc, builder);
}
```

By tracing the ttir generation process, it was found that after passing through TritonReorderBroadcastPass, the “arith.cmpi slt, %1, %cst_1: tensor<1x1xi32>” tensor was converted to “arith.cmpi slt, %0, %c2_i32: i32”.
